src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -152,17 +153,16 @@ class PeerManagerActor(
   private def handleConnection(
       connection: ActorRef,
       remoteAddress: InetSocketAddress,
-      pendingPeers: PeerMap,
-      peers: PeerMap
+      connectedPeers: ConnectedPeers
   ): Unit = {
-    val connectionHandled = isConnectionHandled(remoteAddress, pendingPeers, peers)
-    val isPendingPeersNotMaxValue = pendingPeers.size < peerConfiguration.maxPendingPeers
+    val alreadyConnectedToPeer = connectedPeers.isConnectionHandled(remoteAddress)
```

Comment:
I wonder, sholdn't we check this only based only on `InetAddress` (not `InetSocketAddress`) , as every incoming connection can have different ephemeral port, so the same peer can create several outgoing connections to us from different ports. I remember in other project, I did it as temporarly solution before migrating to indentifing peers here by their `NodeIds`. 

It can have downside, that if there are two peers in the same subnet, we will allow only one of them, but if we will improve it later to identifing peers by node ids, then it can be good temporary solution.

Reply:
>so the same peer can create several outgoing connections to us from different ports

The only consequence from this is that those peers will be detected as repeated only after the handshake, right? Because at least as I was able to test out that might only happen initially, eventually those repeated connections will die our and no new ones will be started

If we had planned to continue improving on this I'd agree with your proposal, but as I'm not sure for how long we'll have to keep that temporary solution I prefer keeping it as is

Reply:
sure, then maybe leave a comment about it here ?